src/test/codegen/issue-75742-format_without_fmt_args.rs

```diff
@@ -0,0 +1,51 @@
+// min-llvm-version: 10.0.0
+// compile-flags: -C opt-level=3
+#![crate_type = "rlib"]
+#![feature(format_args_capture)]
+
+// CHECK-LABEL: @string_from_str
+// CHECK-NOT: __rust_alloc
+// CHECK-NOT: __rust_dealloc
+#[no_mangle]
+pub fn string_from_str() {
+    String::from("a long story");
+    "a long story".to_owned();
+    let _: String = "a long story".into();
+}
+
+// Make sure allocation not happen when result of `format!` is unused and
+// there are no formatting arguments.
+
+// CHECK-LABEL: @format_wo_fmt_args
+// CHECK-NOT: __rust_alloc
+// CHECK-NOT: __rust_dealloc
+#[no_mangle]
+pub fn format_wo_fmt_args() {
+    format!("");
+    format!("a long story");
+    format!("a long story {{");
+}
+
+// CHECK-LABEL: @format_wo_fmt_args_ret
+// CHECK: __rust_alloc
+// CHECK: alloc5alloc18handle_alloc_error
+// CHECK: memcpy
```

Comment:
`CHECK-NOT Argument`, perhaps? 

Some of these tests depend on the symbol mangling format too, which is not great

Reply:
Thanks. Looks like the demangle function is in the IR, no need to use mangling format.